File: scripts/ws/ws.py
"""
File: ws.py
Project: scripts
File Created: Thursday, 16th June 2022 4:35:23 pm
Author: Anonymous (anonymous@gmail.com)
-----
Last Modified: Friday, 2nd February 2024 2:30:15 pm
Modified By: Anonymous (anonymous@gmail.com>)

Run via |
python3 scripts/ws.py --address 'wss://localhost/v1/uploads/{id}?websocket=true&Authorization={token}'
python3 scripts/ws.py --address 'wss://localhost/v1/exports/{id}?websocket=true&Authorization={token}'
python3 scripts/ws.py --address 'wss://localhost/v1/models/{id}?websocket=true&Authorization={token}'
"""

import argparse
import asyncio
import ssl
import time

import websockets

ssl_context = ssl.SSLContext()
ssl_context.check_hostname = False
ssl_context.verify_mode = ssl.CERT_NONE


async def run(uri: str):
    async with websockets.connect(
        uri, ssl=ssl_context, ping_interval=None
    ) as websocket:
        while True:
            try:
                greeting = await websocket.recv()
                print("< {}".format(greeting))
            except Exception as e:
                print(e)
                break

            time.sleep(1)


if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument("--address", type=str, required=True, help="address")

    args = parser.parse_args()

    asyncio.run(run(args.address))

File: scripts/ws/node-ws/ws.js
/*
 * File: ws.js
 * Project: scripts
 * File Created: Tuesday, 9th August 2022 9:53:18 am
 * Author: Anonymous (anonymous@gmail.com)
 * -----
 * Last Modified: Friday, 2nd February 2024 2:33:11 pm
 * Modified By: Anonymous (anonymous@gmail.com>)
 */

import WebSocket from "ws";

const url = "wss://localhost/v1/uploads/62f28a2df68e582a7fb4a27b?websocket=true";
const token = "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjYyZTc2NDBlY2IxYzAyN2NhY2FkNGQ2ZCIsInVpZCI6ImY0MDJlOTE4LTZhNDctNGViZi05NjVmLTNlYTRiMWJmZGRkYSIsImV4cCI6MTY2MDY2NjY3OX0.5_BzuOj_p-4Q6fXvDOyE5Vy4uiy_x2hGfHTO2UpJ4pg";

process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";

const connection = new WebSocket(url, { headers: { "Content-Type": "application/json", Authorization: token } });

connection.on("message", function message(data) {
  console.log("received: %s", data);
});
